### PR TITLE
pcsc-safenet: 10.0.37 -> 10.8.28

### PR DESCRIPTION
**Pull request by charles-dyfis-net**
###### Description of changes

Update pcsc-safenet to a version with better hardware support, clearer licensing, less-unofficial download location. Also updates packaging to deal with `pkgs.openssl` no longer being a 1.x binary.

- Version 10.8 explicitly supports the 5110+ CC line of SafeNet tokens.

- Version 10.8 release notes explicitly document that "from SAC 10.8 release onwards, no license is required for SAC on Linux".

It's not clear to me whether this is sufficient to allow clearing the nonfree flag, but it's certainly an improvement.

Also, we're now getting our downloads from a company that distributes SafeNet hardware rather than a website under the control of an Arch contributor, and thus have a clearer chain-of-custody for these security-critical binaries.

@wldhx, your feedback would be greatly appreciated.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] ~~aarch64-linux~~
  - [ ] ~~x86_64-darwin~~
  - [ ] ~~aarch64-darwin~~
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by wldhx — APPROVED** [review]
Sorry, missed that at the time. I don't work with these tokens anymore so can't test, but looks good at a glance.

Consider picking up the maintainer mantle @charles-dyfis-net!

**Comment by charles-dyfis-net** [comment]
If I wanted to do that, I'd probably want to pick up some older tokens to be able to test against more than one specific (newly-supported!) piece of hardware.

I'll check pricing on eBay to figure out how much doing this responsibly would cost, and mull it over.

File: pkgs/tools/security/pcsc-safenet/default.nix [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
-
-  # https://aur.archlinux.org/packages/sac-core/
-  src = fetchurl {
-    url = "https://storage.spidlas.cz/public/soft/safenet/SafenetAuthenticationClient-core-${version}_amd64.deb";
-    sha256 = "1r9739bhal7ramj1rpawaqvik45xbs1c756l1da96din638gzy5l";
-  };
+  version = "10.8.28";
+
+  # extract debian package from larger zip file
+  src = runCommand "sac.deb" {
```
Comment (RaitoBezarius) [inline pkgs/tools/security/pcsc-safenet/default.nix:18]: Why `fetchzip` is not enough here?

File: pkgs/tools/security/pcsc-safenet/default.nix [hunk 2]
```diff
@@ -41,35 +50,34 @@ stdenv.mkDerivation rec {
...
+
+    mkdir "$out"
+    cp -r ./* "$out/"
+
+    (
+      cd "$out/lib/" || exit
+      for f in *.so.*.*.*; do
+        ln -sf "$f" "''${f%.*}" || exit
+        ln -sf "$f" "''${f%.*.*}" || exit
+        ln -sf "$f" "''${f%.*.*.*}" || exit
```
Comment (RaitoBezarius) [inline pkgs/tools/security/pcsc-safenet/default.nix:67]: It's quite complicated, please add comments to explain how it works, what is it supposed to do, etc.

File: pkgs/tools/security/pcsc-safenet/default.nix [hunk 3]
```diff
@@ -26,13 +35,13 @@ stdenv.mkDerivation rec {
...
-    gtk2
-    openssl
+    gtk3
+    openssl_1_1
     pcsclite
   ];
 
   runtimeDependencies = [
-    openssl
+    openssl_1_1
```
Comment (RaitoBezarius) [inline pkgs/tools/security/pcsc-safenet/default.nix:44]: Hm, we want to kill OpenSSL 1.1, is there a big reason for why OpenSSL3 wouldn't work?

**Review by RaitoBezarius — CHANGES_REQUESTED** [review]
Some general comments, I'm convinced by the high level stuff, just need some comments.

File: pkgs/tools/security/pcsc-safenet/default.nix (hunk 1 above)
Comment (charles-dyfis-net) [reply pkgs/tools/security/pcsc-safenet/default.nix:18]: The only argument against fetchzip is that this way we only unpack the one file we need, and not the others we don't. If the extra size of the intermediate build outputs isn't too much of a cost to pay for standardization, I'll make that change.

File: pkgs/tools/security/pcsc-safenet/default.nix (hunk 3 above)
Comment (charles-dyfis-net) [reply pkgs/tools/security/pcsc-safenet/default.nix:44]: Good questions. The docs imply it _should_ be supported, but in practice it very much does not. Getting a closed-source binary blob, unfortunately, it's difficult to dig further.

Comment (RaitoBezarius) [reply pkgs/tools/security/pcsc-safenet/default.nix:42]: Got it! Keep in mind that https://www.openssl.org/blog/blog/2023/03/28/1.1.1-EOL/ :).

**Review by RaitoBezarius — APPROVED** [review]
LGTM!

File: pkgs/tools/security/pcsc-safenet/default.nix (hunk 1 above)
Comment (RaitoBezarius) [reply pkgs/tools/security/pcsc-safenet/default.nix]: Yes, I think, we already have the ZIP, so it does not cost us more to do a fetchzip.

**Pull request merged**